cli/crud/import/containers.go

```diff
@@ -0,0 +1,163 @@
+	Progress bool
+	// Operation sets type of crud stored procedures during import (inser of replace).
+	Operation string
+	// OnError sets the action when an error is detected (stop or skip).
+	OnError string
+	// NullVal sets null interpretation for input data.
+	NullVal string
+}
+
+// dumpSubsystemFdСontainer contains file descriptors for the disk dump subsystem.
+type dumpSubsystemFdСontainer struct {
+	// logFile is file descriptor of file with logs about unsuccessful imported data.
+	logFile *os.File
+	// errorFile is file descriptor of file with unsuccessful imported data.
+	errorFile *os.File
+	// successFile is file descriptor of file with successful imported data.
+	successFile *os.File
+	// progressFile is file descriptor of file with progress information.
+	progressFile *os.File
+}
+
+// parserCtxСontainer contains a part of the tuple context that related to the parser.
+type parserCtxСontainer struct {
+	// CsvRecordPosition is current position in input csv file.
+	CsvRecordPosition uint32 `yaml:"csvRecordPosition" json:"csvRecordPosition"`
+	// UnparsedCsvRecord is current unparsed csv record.
+	UnparsedCsvRecord string `yaml:"unparsedCsvRecord" json:"unparsedCsvRecord"`
+	// ParsedSuccess indicates the success of last parsing iteration.
+	ParsedSuccess bool `yaml:"parsedSuccess" json:"parsedSuccess"`
+	// ParsedCsvRecord contains parsed result of last parsing iteration.
+	ParsedCsvRecord []string `yaml:"parsedCsvRecord" json:"parsedCsvRecord"`
+	// ErrorMsg contains error description of last parsing iteration.
+	ErrorMsg string `yaml:"errorMsg" json:"errorMsg"`
+}
+
+// parserCtxСontainer contains a part of the tuple context that related to the router.
+type crudCtxСontainer struct {
+	// TupleImported indicates the success of the import.
+	TupleImported bool `yaml:"tupleImported" json:"tupleImported"`
+	// CastedTuple сontains a set of converted (on router side) fields to insert into space.
+	CastedTuple []interface{} `yaml:"castedTuple"  json:"castedTuple"`
+	// CrudErr сontains error description from crud stored procedure.
+	CrudErr string `yaml:"crudErr" json:"crudErr"`
+}
+
+// batchTupleСontainer contains the context of a tuple within a batch.
+type batchTupleСontainer struct {
+	// TupleNumber contains the number of the tuple within the batch.
+	// It can take values from the range [1, batchSize] for initialized tuples within a batch.
+	// For uninitialized tuples within a batch, a 0 value is used.
+	// The current number of initialized tuples is recorded in struct batchСontainer->TuplesAmount.
+	TupleNumber uint32 `yaml:"tupleNumber" json:"tupleNumber"`
+	// ParserCtx contains parser context related to this tuple.
+	ParserCtx parserCtxСontainer `yaml:"parserCtx" json:"parserCtx"`
+	// CrudCtx contains router side context related to this tuple.
+	CrudCtx crudCtxСontainer `yaml:"crudCtx" json:"crudCtx"`
+}
+
+// batchСontainer contains the context of the batch, including necessary meta information for crud.
```

Comment:
The `BatchContainer' and `BatchTruleContainer' are very close names. I suggest you think about the names.